R/tar_insuvimeh.R

```diff
@@ -0,0 +1,123 @@
+catalogue_insuvimeh_files <-  function(gdb){
+  fps <- list.files(gdb,
+                    full.names =T,
+                    recursive = T,
+                    pattern = "\\d{4}.nc$")
+  fps_w_parents <- list.files(gdb,
+                              recursive = T,
+                              pattern = "\\d{4}.nc$")
+  lr <- fps %>% 
+    map2(fps_w_parents,\(fp_nc,fp_par){
+      
+      fp_base <- basename(fp_nc)
+      # meta data collection  - to write band_name as "{pub_date}.lt_{leadtime}"
+      valid_time_meta <- ncmeta::nc_att(fp_nc,"T","units")$value$units
+      start_time_meta <- ncmeta::nc_att(fp_nc,"S","units")$value$units
+      valid_date <- as_date(str_extract(valid_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      start_date <- as_date(str_extract(start_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      lead_time_diff <- interval(start_date,valid_date)
+      lt <- lead_time_diff %/% months(1)
+      tibble(
+        file_name_parent = fp_par,
+        file_name= fp_base,
+        pub_date = start_date,
+        valid_date =valid_date,
+        lt = lt
+      )
+    }) %>% 
+    list_rbind()
+  return(lr)
+}
+
+
+load_insuvimeh_raster <-  function(gdb){
+  fps <- list.files(gdb,
+                    full.names =T,
+                    recursive = T,
+                    pattern = "\\d{4}.nc$")
+  
+  lr <- fps %>% 
+    map(\(fp_nc){
+      
+      # meta data collection  - to write band_name as "{pub_date}.lt_{leadtime}"
+      valid_time_meta <- ncmeta::nc_att(fp_nc,"T","units")$value$units
+      start_time_meta <- ncmeta::nc_att(fp_nc,"S","units")$value$units
+      valid_date <- as_date(str_extract(valid_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      start_date <- as_date(str_extract(start_time_meta,"\\d{4}-\\d{2}-\\d{2}"))
+      lead_time_diff <- interval(start_date,valid_date)
+      lt <- lead_time_diff %/% months(1)
+      bname<- paste0(start_date,".lt_",lt)
+      
+      # Now we open the each file and turn it into a terra::rast() 
+      cat(bname,"\n")
+      dat <- ncdf4::nc_open(fp_nc)
+      lon <- ncdf4::ncvar_get(dat,"X")
+      lat <- ncdf4::ncvar_get(dat,"Y")
+      value_array <- ncdf4::ncvar_get(dat,"deterministic")
+      value_array_fixed <- aperm(value_array,c(2,1))
+      
+      rtmp <- terra::rast(
+        x= value_array_fixed,
+        extent =ext(
+          min(lon),
+          max(lon),
+          min(lat),
+          max(lat)
+        ),
+        crs= "OGC:CRS84"
+      )
+      # set correct band nanme
+      rtmp %>% 
```

Comment:
Is this being saved? Does `set.names()` affect it in place?

Reply:
yeah kind of bizarre, but it does